lib/rules/no-warning-comments.js

```diff
@@ -64,59 +64,31 @@ module.exports = {
          */
         function convertToRegExp(term) {
             const escaped = escapeRegExp(term);
-            const wordBoundary = "\\b";
-            const eitherOrWordBoundary = `|${wordBoundary}`;
-            let prefix;
 
             /*
-             * If the term ends in a word character (a-z0-9_), ensure a word
-             * boundary at the end, so that substrings do not get falsely
-             * matched. eg "todo" in a string such as "mastodon".
-             * If the term ends in a non-word character, then \b won't match on
-             * the boundary to the next non-word character, which would likely
-             * be a space. For example `/\bFIX!\b/.test('FIX! blah') === false`.
-             * In these cases, use no bounding match. Same applies for the
-             * prefix, handled below.
+             * When matching at the start, ignore leading whitespace, and
+             * there's no need to worry about word boundaries.
+             *
+             * These expressions for the prefix and suffix are designed as follows:
+             * (^) and ($) handle any terms at the beginning/end of a line, regardless of start/end characters
+             * ([^\w]) handle terms starting/ending with non-word characters preceded/followed by other non-word characters.
+             *      e.g. terms: ["!TODO", "FIX!"] matches "!!!TODO blah" or "FIX!!! blah"
+             * (\b) handles terms starting with word characters preceded/followed by word boundary
+             *      e.g. terms: ["TODO"] matches "!TODO blah" or "blah TODO!", but not "mastodon"
```

Comment:
All 3 are needed. I just forgot to include the tests for the cases that fail when `\b` is omitted.

For terms `["!FIX", "FIX!"]`, the following comments would fail without `\b`:

```js
// FIX!comment
// comment!FIX
```

I will add new tests for this.

Reply:
I will also revise this explanatory comment. It looks like I confused myself as I wrote it, and it's wrong on several points.